docs/debugging/providers_and_languages.md

```diff
@@ -0,0 +1,27 @@
+# Providers and Languages
+
+Languages and providers run as their own executables, so debugging isn't as
+simple as attaching to the engine.
+
+Instead you'll have to launch them in a debugger and provide the information
+for connecting to them to the engine via environmental variables.
+
+## Languages
+
+Languages work very similary to [Providers](#Providers).
+
+Instead of PULUMI_DEBUG_PROVIDERS environmental variable, you use PULUMI_DEBUG_LANGUAGES like so:
+
+```sh
+PULUMI_DEBUG_LANGUAGES="go:12345" pulumi preview
+```
+
+After opening your language executable (eg `pulumi-language-go`) in your debugger of choice.
+
+### Internals
+
+This works _slightly_ differently from PULUMI_DEBUG_PROVIDERS.  Providers don't
+need an argument passed of the address of the server, but languages do.  To get
+around this, we have introduced a special case of launching a language provider
+which will wait for the engine address to be sent with the special
+[](pulumirpc.LanguageRuntime.Attach) rpc.
```

Comment:
This is probably outdated now. Can jettison docs and return to them later if easier.